Userland/Libraries/LibCore/Group.cpp

```diff
@@ -48,8 +48,10 @@ ErrorOr<void> Group::add_group(Group& group)
     if (!file)
         return Error::from_errno(errno);
 
-    if (putgrent(&gr, file) < 0)
+    if (putgrent(&gr, file) < 0) {
+        fclose(file);
```

Comment:
Since we always want to close the file on error or success, we can just use a scope guard declared right after we open the file. Which will close the file when the scope exits, no matter what. 

```cpp
#include <AK/ScopeGuard.h>

... snip ... 

ScopeGuard file_guard = [file] { fclose(file); };
```

Then we can get rid of both of thee other `fclose` calls.